services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/table/MaterializedViewManagerImpl.java

```diff
@@ -346,24 +354,24 @@ void createOrRebuildViewHoldingWriteLockAndAllDependentReadLocks(QueryTranslator
 		LOG.info("Materialized view " + idAndVersion + " set to AVAILABLE");
 	}
 	
-	private IdAndVersion[] getAvailableDependentIds(QueryTranslator sqlQuery) {
+	/**
+	 * Converts the provided list to an array if each dependency's status is AVAILABLE.
+	 * 
+	 * @param dependentTables
+	 * @return {@link Optional#empty()} when one or more dependency status is not AVAILABLE.
+	 */
+	Optional<IdAndVersion[]> getAvailableDependentIds(List<IdAndVersion> dependentTables) {
 		// Check if each dependency is available. Note: Getting the status of a
 		// dependency can also trigger it to update.
-		List<IdAndVersion> dependentTables = sqlQuery.getTableIds();
 		for (IdAndVersion dependent : dependentTables) {
-			TableStatus status = tableManagerSupport.getTableStatusOrCreateIfNotExists(dependent);
-			switch (status.getState()) {
-			case AVAILABLE:
-				break;
-			case PROCESSING:
-				throw new RecoverableMessageException();
-			case PROCESSING_FAILED:
-				throw new IllegalArgumentException("Cannot build materialized view " + sqlQuery.getIndexDescription().getIdAndVersion() + ", the dependent table " + dependent + " failed to build");
-			default:
-				throw new IllegalStateException("Cannot build materialized view " + sqlQuery.getIndexDescription().getIdAndVersion() + ", unsupported state for dependent table " + dependent + ": " + status.getState());
+			TableState state = tableManagerSupport.getTableStatusOrCreateIfNotExists(dependent).getState();
```

Comment:
What happens if a dependency is in PROCESSING_FAILED and the view was never build?